source/Nevermore/Advanced/ReadTransaction.cs

```diff
@@ -810,6 +823,16 @@ void ITransactionDiagnostic.WriteCurrentTransactions(StringBuilder output)
             registry.WriteCurrentTransactions(output);
         }
 
+        DbTransaction BeginTransaction(IsolationLevel isolationLevel, string sqlServerTransactionName)
+        {
+            if (connection is SqlConnection sqlConnection)
+            {
+                return sqlConnection.BeginTransaction(isolationLevel, sqlServerTransactionName);
```

Comment:
@adam-mccoy this here (`DbConnection` not having a named `BeginTransaction` overload) was why I made the ISqlConnection interface rather than passing DbConnection around the place. I don't disagree with your change here (it felt like a bit of a coin-toss when I made the interface) but it'd be interesting to get your thoughts about why